### PR TITLE
Syntax highlighting

### PR DESCRIPTION
**Pull request by fox0430**
Still in development.

### CODE REVIEW

File: src/moepkg/highlight.nim [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    of gtWhitespace:
+      var numOfSpace = 0
+      for i in 0 ..< str.len:
+        if str[i] != '\n':
+          numOfSpace.inc
+      color = concat(color, defaultColor.repeat(numOfSpace))
+    else:
+      color = concat(color, defaultColor.repeat(str.len))
+
+  var splitBuffer: seq[string] = @[]
```
Comment (walkre-niboshi) [inline src/moepkg/highlight.nim]: `let splitedBuffer = splitLines`

File: src/moepkg/highlight.nim [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  for buffer in buffer.splitLines:
+    splitBuffer.add(buffer)
+
+  result = @[]
```
Comment (walkre-niboshi) [inline src/moepkg/highlight.nim]: Unnecessary initialization.

File: src/moepkg/highlight.nim [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  var all = 0
+  for i in 0 ..< splitBuffer.len:
+    var line: seq[Colorpair] = @[]
```
Comment (walkre-niboshi) [inline src/moepkg/highlight.nim]: Unnecessary initialization.

File: src/moepkg/highlight.nim [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    for j in 0 ..< splitBuffer[i].len:
+      if splitBuffer[i].len != 0:
```
Comment (walkre-niboshi) [inline src/moepkg/highlight.nim]: When throughing this line, splitBuffer[i].len isn't equal to 0.
So this condition is meaningless .

**Comment by fox0430** [comment]
It's not bad with release build.

File: src/moepkg/highlight.nim [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+proc setHighlightInfo*(buffer: GapBuffer[seq[Rune]], language: string, setting: bool): seq[seq[Colorpair]] =
+  if setting and language != "Plain":
+    return getHighlightColor($buffer, language)
+  else:
+    return getAllDefaultColor($buffer)
+
+proc initLanguage*(filename: string): string =
+  
+  let fileExtension = filename.split('.')
```
Comment (walkre-niboshi) [inline src/moepkg/highlight.nim]: `let extention = filename.splitFile.ext`

Comment (fox0430) [reply src/moepkg/highlight.nim]: Good idea.

**Comment by walkre-niboshi** [comment]
Crash when inserting into an empty text.

**Pull request merged**